src/order-actions/reject-order.component.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useCallback } from "react";
+import { OverflowMenuItem, OverflowMenu } from "@carbon/react";
+import { useTranslation } from "react-i18next";
+import { showModal } from "@openmrs/esm-framework";
+import { Result } from "../work-list/work-list.resource";
+
+interface RejectOrderOverflowMenuItemProps {
+  order: Result;
+}
+
+const RejectOrderOverflowMenuItem: React.FC<
+  RejectOrderOverflowMenuItemProps
+> = ({ order }) => {
+  const { t } = useTranslation();
+
+  const handleRejectOrder = useCallback(() => {
+    const dispose = showModal("reject-order-dialog", {
+      closeModal: () => dispose(),
+      order,
+    });
+  }, [order]);
+  return (
+    <OverflowMenuItem
+      itemText={t("rejectOrder", "Reject Order")}
+      onClick={handleRejectOrder}
```

Comment:
Adding a className and defining some stlying would suffice but it looks optional though.